File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int first = 0;
        int last = height.length - 1;
        int area = 0;
        
        while (last > first) {
            area = Math.max(area, Math.min(height[first], height[last]) * Math.abs(first - last));
            
            if (height[first] < height[last]) {
                first++;
            } else {
                last--;
            }
        }
        
        return area;
    }
}

File: 11-container-with-most-water/container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */

var maxArea = function (height) {
  let first = 0;
  let last = height.length - 1;
  let area = 0;

  while (last > first) {
    const currentHeight = Math.min(height[first], height[last]);
    const width = last - first;
    area = Math.max(area, currentHeight * width);

    if (height[first] < height[last]) {
      first++;
    } else {
      last--;
    }
  }

  return area;
};
